DataStorageEngine/DSECommonSenseProxy.h

```diff
@@ -0,0 +1,169 @@
+ 
+	Error handling is done through NSError objects. Server side error codes will be directly copied into the error. Client side errors have their own codes specified in DSEErrorCodes.h. The userInfo object will contain an error message if applicable, or will be nil otherwise. All DataStorageEngine will use the error domain "nl.sense.DataStorageEngine.ErrorDomain".
+ 
+ */
+@interface DSECommonSenseProxy : NSObject {
+	NSString *appKey;					//The app key
+	NSString *urlBase;					//The base url to use, will differ based on whether to use live or staging server
+	NSString *urlAuth;				//The base url to use for authentication, will differ based on whether to use live or staging server
+}
+
+
+/**
+ Default initializaler for the DSECommonSenseProxy.
+ 
+ Takes an app key that will be used throughout the proxies lifetime. Needs to know whether to talk to the live server or the staging server. This cannot be changed during the proxies lifetime. If you need to change this you simply init a new commonsense proxy.
+ 
+ @param useLiveServer	If YES, the live server will be used. If NO, the staging server will be used.
+ @param theAppKey		An application key that identifies the application to the commonsense server. Cannot be empty.
+ @result				Initialized DSECommonSenseProxy
+*/
+- (id) initForLiveServer: (BOOL) useLiveServer withAppKey: (NSString *) theAppKey;
+
+#pragma mark User
+/**
+ @name Users
+ */
+
+/**
+ Login a user
+ 
+ @note When a user is already logged in, this call will still succeed and return a new sessionID.
+ 
+ @param username		A user account in commonsense is uniquely identified by a username. Cannot be empty.
+ @param password		A password in commonsense does not have any specific requirements. It will be MD5 hashed before sending to the server so the user does not have to provide a hashed password. Cannot be empty.
+ @param error			Reference to an NSError object that will contain error information if an error occurs. Cannot be nil.
+ @return				Session ID. Will be nil if the call fails.
+ */
+- (NSString *) loginUser: (NSString *) username andPassword: (NSString *) password andError: (NSError **) error;
+
+/**
+ Logout the currently logged in user.
+ 
+ @param sessionID		The sessionID of the user to logout. Cannot be empty.
+ @param error			Reference to an NSError object that will contain error information if an error occurs. Cannot be nil.
+ @return				Whether or not the logout finished succesfully.
+ */
+- (BOOL) logoutCurrentUserWithSessionID: (NSString *) sessionID andError: (NSError **) error;
+
+
+//- (BOOL) deleteCurrentUserWithSessionID: (NSString *) sessionID andError: (NSError **) error;
+
+#pragma mark Sensors and Devices
+/**
+ @name Sensors and Devices (DSESource backend equivalent)
+*/
+
+
+/**
+ Create a new sensor in the commonsense backend
+ 
+ Each sensor in commonsense is uniquely identified by a name and devicetype combination. If this combination already exists in the commonsense backend this call will fail.
```

Comment:
@jorishjanssen This is not true for CommonSense, and we don't want this since you can have multiple sensors with the same properties connected to a different device